src/Validation.hs

```diff
@@ -0,0 +1,395 @@
+{- |
+Copyright:  (c) 2014 Chris Allen, Edward Kmett
+            (c) 2018-2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Lightweight pure validation based on 'Applicative' and 'Selective' functors.
+
+'Validation' is a monoidal sibling to 'Either' but 'Validation' doesn't have a
+'Monad' instance. 'Validation' allows to accumulate all errors instead of
+short-circuting on the first error so you can display all possible errors at
+once.
+
+Common use-cases include:
+
+1. Validating each input of a form with multiple inputs.
+2. Performing multiple validations of a single value.
+
+Instances of different standard typeclasses provide various semantics:
+
+1. 'Functor': change the type inside 'Success'.
+2. 'Bifunctor': change both 'Failure' and 'Success'.
+3. 'Applicative': apply function to values inside 'Success' and accumulate
+   errors inside 'Failure'.
+4. 'Semigroup': accumulate both 'Failure' and 'Success' with '<>'.
+5. 'Monoid': 'Success' that shores 'mempty'.
+6. 'Alternative': return first 'Success' or accumulate all errors inside 'Failure'.
+-}
+
+{-# LANGUAGE CPP                  #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Validation
+       ( -- * How to use
+         -- $use
+         Validation(..)
+       , validationToEither
+       , eitherToValidation
+       ) where
+
+import Control.Applicative (Alternative (..), Applicative (..))
+import Data.Bifunctor (Bifunctor (..))
+import Data.Foldable (Foldable (..))
+import Data.Kind (Constraint)
+import GHC.TypeLits (ErrorMessage (..), TypeError)
+
+#if MIN_VERSION_base(4,10,0)
+import Data.Bifoldable (Bifoldable (..))
+import Data.Bitraversable (Bitraversable (..))
+#endif
+
+-- >>> $setup
+-- >>> import Relude
+
+{- $use
+
+Take for example a type @Computer@ that needs to be validated:
```

Comment:
```suggestion
To give you an example of usage of 'Validation' let's take a @Computer@  data type that needs to be validated:
```

Reply:
Maybe give more description of what `Computer` is, as it is not that obvious froom the name how can computer be validated and why it needs it?